src/apps/allbridge/positions.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  PositionsHook,
+  AppTokenPositionDefinition,
+  UnknownAppTokenError,
+  TokenDefinition,
+  ContractPositionDefinition,
+} from '../../types/positions'
+import { Address } from 'viem'
+import { DecimalNumber, toDecimalNumber } from '../../types/numbers'
+import BigNumber from 'bignumber.js'
+import { getClient } from '../../runtime/client'
+import { getTokenId } from '../../runtime/getTokenId'
+import { getAllbridgeTokenInfo } from './api'
+import { poolAbi } from './abis/pool'
+
+const ALLBRIDGE_LOGO =
+  'https://raw.githubusercontent.com/valora-inc/dapp-list/main/assets/allbridgecore.png'
+
+const hook: PositionsHook = {
+  getInfo() {
+    return {
+      id: 'allbridge',
+      name: 'Allbridge',
+      description: '',
+    }
+  },
+  async getPositionDefinitions(networkId, address) {
+    const allbridgeTokenInfo = (await getAllbridgeTokenInfo({ networkId }))
+      .tokens
+
+    const client = getClient(networkId)
+
+    const balances = await Promise.all(
+      allbridgeTokenInfo.map(async (tokenInfo) => {
+        return address
+          ? client.readContract({
+              address: tokenInfo.poolAddress,
+              abi: poolAbi,
+              functionName: 'balanceOf',
+              args: [address as Address],
+            })
+          : undefined
+      }),
+    )
+
+    const rewards = await Promise.all(
+      allbridgeTokenInfo.map(async (tokenInfo) => {
+        return address
+          ? client.readContract({
+              address: tokenInfo.poolAddress,
+              abi: poolAbi,
+              functionName: 'pendingReward',
+              args: [address as Address],
+            })
+          : undefined
+      }),
+    )
+
+    return allbridgeTokenInfo.flatMap((tokenInfo, i) => {
+      const apr = new BigNumber(tokenInfo.apr7d).toNumber() * 100
+
+      const balanceOf = balances[i]
+      const pendingReward = rewards[i]
+
+      const useAToken = !address || (!!balanceOf && balanceOf > 0)
+      const showReward = !!pendingReward && pendingReward > 0
+      return [
```

Comment:
half-hearted nit: I think this the below is easier to read.  maybe I'm old-fashioned, so take it or leave it.

```ts
const definitions: PositionDefinition = []
if (useAToken) {
  definitions.push({...})
}
if (showReward) {
  definitions.push({...})
}
return definitions
````

Reply:
I'm indifferent either way, I had it as is since that matches the Aave implementation, might be good to have things standardized somewhat so as we add more pools things look similar / easy to understand across pool providers.